dotnet-grad/Controllers/UserController.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using dotnet_grad.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace dotnet_grad.Controllers
+{
+  [ApiController]
+  [Route("api/[controller]")]
+  public class UserController : ControllerBase
+  {
+    TestContext testContext;
+    public UserController(TestContext _testContext)
+    {
+      testContext = _testContext;
+    }
+
+    [HttpGet("GetAllUsers")]
+    public List<UserModel> getUsers()
+    {
+      return testContext.Users.ToList();
+    }
+
+    [HttpGet("{id}")]
+    public ActionResult<UserModel> getUser(int id)
+    {
+      try
+      {
+        UserModel user = testContext.Users.Find(id);
+        return Ok(user);
+      }
+      catch (Exception e)
+      {
+        return NotFound("user not found");
```

Comment:
How sure are you that any exception being thrown by `Users.Find` relate to the specific user not existing within the DB? what about a DB connection error, or a timeout exception?